testkit/src/main/scala/com/lightbend/lagom/javadsl/testkit/ServiceTest.scala

```diff
@@ -199,9 +200,10 @@ object ServiceTest {
     val b2 =
       if (setup.persistence) {
         val cassandraPort = CassandraLauncher.randomPort
-        val cassandraDirectory = new File("target/" + testName)
+        val cassandraDirectory = Files.createTempDirectory(testName).toFile
+        FileUtils.deleteRecursiveOnExit(cassandraDirectory)
         val t0 = System.nanoTime()
-        CassandraLauncher.start(cassandraDirectory, CassandraLauncher.DefaultTestConfigResource, clean = true, port = 0)
+        CassandraLauncher.start(cassandraDirectory, CassandraLauncher.DefaultTestConfigResource, clean = false, port = 0)
```

Comment:
Because if the directory doesn't exist, it throws an exception. The launcher attempts to create the directory after attempting to delete the directory, the directory deletion fails if the directory didn't already exists.  I'm not sure how or if this has ever worked - I think in our tests we create the directory first, I don't think we should expect end users to do that.

The other problem with the old code was that it used the target directory is the working directory - but in sbt, and in general, you shouldn't depend on the current working directory in tests, in some cases the target directory would be the parent projects target directory.  So, what I did was to use the Java 7 Files.createTemporaryDirectory to obtain a directory.  Since this is guaranteed to be a new empty directory, there's no need to clean it first.